File: src/main/java/com/github/glassmc/loader/launch/Launcher.java
package com.github.glassmc.loader.launch;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Launcher {

    private String target = null;
    private List<String> arguments = new ArrayList<>();
    private List<ITransformer> transformers = new ArrayList<>();

    public void launch() {
        ClassLoader classLoader = new Loader(transformers);

        try {
            Class<?> wrapperClass = classLoader.loadClass("com.github.glassmc.loader.launch.Wrapper");
            Method launchMethod = wrapperClass.getMethod("launch", String.class, String[].class);
            launchMethod.invoke(null, target, arguments.toArray(new String[0]));
        } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
            e.printStackTrace();
        }
    }

    public void setTarget(String target) {
        this.target = target;
    }

    public void addArguments(String... arguments) {
        this.arguments.addAll(Arrays.asList(arguments));
    }

}

File: core/src/main/kotlin/ml/glassmc/loader/launch/Launcher.kt
package ml.glassmc.loader.launch

class Launcher {

    lateinit var target: String
    val arguments: MutableList<String> = ArrayList()
    val transformers: MutableList<ITransformer> = ArrayList()

    fun launch() {
        val classLoader = Loader(transformers)

        val `class` = classLoader.loadClass("ml.glassmc.loader.launch.Wrapper")
        val method = `class`.getMethod("launch", String::class.java, Array<String>::class.java)
        method.invoke(`class`.newInstance(), target, arguments)
    }

}